packages/sdk-cli/src/models/localRelay/launch.test.ts

```diff
@@ -28,74 +29,44 @@ describe('launch', () => {
     }
   });
 
-  it('spawns a process and logs output to a log file', (done) => {
-    const logOutput = 'test output';
-    logFilePath = join(__dirname, './launch-test-output.txt');
-    const logFile = createWriteStream(logFilePath);
-
-    // https://stackoverflow.com/a/44846808/6539857
-    // Without 'open' event spawn() won't accept the WriteStream, because
-    // "[log stream] must have an underlying descriptor (file streams do not until the 'open' event has occurred)"
-    // Related: https://github.com/nodejs/node-v0.x-archive/issues/4030
-    logFile.on('open', (fd) => {
-      const nodeArgs = ['-e', `console.log('${logOutput}')`];
-      subprocess = launch(nodeArgs, fd);
-
-      subprocess.on('error', (error) => {
-        return done(error);
-      });
-
-      subprocess.on('close', async () => {
-        try {
-          await expect(
-            fsPromises.readFile(logFilePath, { encoding: 'utf8' }),
-          ).resolves.toMatch(logOutput);
-        } catch (error) {
-          return done(error);
-        }
-
-        return done();
-      });
-    });
-
-    logFile.on('error', (error) => {
-      return done(error);
-    });
-  });
-
-  // TODO: Separate log and error files?
-  it('spawns a process and logs error to a log file', (done) => {
-    const logOutput = 'test output';
-    logFilePath = join(__dirname, './launch-test-error-output.txt');
-    const logFile = createWriteStream(logFilePath);
-
-    // https://stackoverflow.com/a/44846808/6539857
-    // Without 'open' event spawn() won't accept the WriteStream, because
-    // "[log stream] must have an underlying descriptor (file streams do not until the 'open' event has occurred)"
-    // Related: https://github.com/nodejs/node-v0.x-archive/issues/4030
-    logFile.on('open', (fd) => {
-      const nodeArgs = ['-e', `console.log('${logOutput}')`];
-      subprocess = launch(nodeArgs, fd);
-
-      subprocess.on('error', (error) => {
-        return done(error);
+  it.each<[string, keyof typeof console, jest.DoneCallback?]>([
+    ['output', 'log'],
+    ['error', 'error'],
+  ])(
+    'spawns a process and logs %s to a log file',
+    (outputType, consoleMethodName, done) => {
+      const logOutput = `test ${outputType}`;
+      logFilePath = join(__dirname, `./launch-test-${outputType}.txt`);
+      const logFile = createWriteStream(logFilePath);
+
+      // https://stackoverflow.com/a/44846808/6539857
+      // Without 'open' event spawn() won't accept the WriteStream, because
+      // "[log stream] must have an underlying descriptor (file streams do not until the 'open' event has occurred)"
+      // Related: https://github.com/nodejs/node-v0.x-archive/issues/4030
+      logFile.on('open', (fd) => {
+        const nodeArgs = ['-e', `console.${consoleMethodName}('${logOutput}')`];
+        subprocess = launch(nodeArgs, fd);
+
+        subprocess.on('error', (error) => {
+          return done!(error);
```

Comment:
Does this ever get fired? Both cases look happy path so I don't think so?

Reply:
I'm not saying that to remove it, I can see you're trying to catch and fail the test if something goes wrong. Just hard to know it works as intended without it actually running.

Reply:
You would have to have a bad executable path or similar to cause this.

Reply:
Nope, I've tried it before and all the following things didn't work (i.e. they either output the error to subprocess.stderr, or didn't have an effect at all):

https://nodejs.org/api/child_process.html#event-error

For option 1:
- Pass bad executable path: `node randomrandom`
- Pass bad option/parameter: `node --randomradnom`

For option 2:
- `subprocess.kill('INVALID_SIGNAL')`

For option 3:
- `subprocess.send()` after subprocess closed or finished

Reply:
Interesting, leave as is then.

Reply:
Searching for "emit('error'" on https://github.com/nodejs/node/blob/master/lib/internal/child_process.js, it seems like `'error'` is reserved for only very corner cases, majority of other erroneous cases are handled directly, without emitting 'error'. I.e. it happens if you set a IPC channel on your spawned process, then disconnect the channel and send a message – will emit an error telling you that you can't deliver messages over a closed IPC channel.